### PR TITLE
Minor tweaks to get things working

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: src/main/java/com/okta/developer/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         // @formatter:off
         http
             .csrf()
-            .ignoringAntMatchers("/h2-console/**")
+            .ignoringAntMatchers("/h2-console/**", "/scim/**")
```
Comment (bdemers) [inline src/main/java/com/okta/developer/config/SecurityConfiguration.java:61]: CSRF was blocking Okta from calling the /scim endpoints

File: src/main/java/com/okta/developer/security/oauth2/ScimUserService.java [hunk 2]
```diff
@@ -107,6 +107,7 @@ public ScimUser create(ScimUser resource) throws UnableToCreateResourceException
         if (user.isPresent()) {
             throw new UnableToCreateResourceException(Response.Status.CONFLICT, "User '" + resource.getUserName() + "' already exists.");
         } else {
+            resource.setId(UUID.randomUUID().toString()); // TODO: this should probably be configured in the persistence layer?
```
Comment (bdemers) [inline src/main/java/com/okta/developer/security/oauth2/ScimUserService.java:110]: This is 🤮, but I didn't want to dig into this, I'm guessing you could tweak it.
The SCIM server doesn't set an ID on create, so the SCIMple server will need to do it.

File: src/main/java/com/okta/developer/security/oauth2/ScimUserService.java [hunk 3]
```diff
@@ -132,6 +133,7 @@ private ScimUser saveUserFromIdP(ScimUser scimUser) {
         User user = new User();
         user.setId(scimUser.getId());
         user.setEmail(scimUser.getPrimaryEmailAddress().get().getValue());
+        user.setLogin(user.getEmail().toLowerCase().toLowerCase(Locale.ROOT));
```
Comment (bdemers) [inline src/main/java/com/okta/developer/security/oauth2/ScimUserService.java:136]: There is an ugly line below that calls `getUserWithAuthoritiesByLogin` and uses this `login` value.  The OIDC user ID isn't available in the SCIM payload 😢

File: src/main/java/com/okta/developer/web/rest/AccountResource.java [hunk 4]
```diff
@@ -46,7 +48,15 @@ public AccountResource(UserService userService) {
...
     @SuppressWarnings("unchecked")
     public AdminUserDTO getAccount(Principal principal) {
-        Optional<User> user = userService.getUserWithAuthoritiesByLogin(principal.getName());
+        // FIXME: really ugly line, could cause NPE and class cast exceptions
+        String email =
+            ((OAuth2AuthenticationToken) principal).getPrincipal()
+                .getAttributes()
+                .getOrDefault("email", "")
+                .toString()
+                .toLowerCase(Locale.ROOT);
```
Comment (bdemers) [inline src/main/java/com/okta/developer/web/rest/AccountResource.java]: error-prone line 😢

File: src/main/resources/config/application.yml [hunk 5]
```diff
@@ -161,6 +161,8 @@ spring:
             client-secret: web_app
             scope: openid, profile, email, offline_access # last one for refresh tokens
 
+  jersey:
+    application-path: /scim
```
Comment (bdemers) [inline src/main/resources/config/application.yml:165]: This moves all of the JAX-RS SCIMple endpoints under /scim, and lets the _normal_ Spring Controllers work as expected:

Something like this is also needed when configuration Okta's SCIM simple with Basic Auth:

```yaml
spring:
  security:
    user:
      name: <username>
      password: <password>
```

File: src/main/java/com/okta/developer/security/oauth2/ScimUserService.java (hunk 2 above)
Comment (mraible) [reply src/main/java/com/okta/developer/security/oauth2/ScimUserService.java:110]: In the previous implementation, I used the "sub" claim, but I guess that's not available with SCIM.

File: src/main/java/com/okta/developer/web/rest/AccountResource.java (hunk 4 above)
Comment (mraible) [reply src/main/java/com/okta/developer/web/rest/AccountResource.java]: `SecurityUtils.getCurrentUserLogin()` might be easier.

**Comment by mraible** [comment]
I've tried this branch and I'm unable to get things working from the Okta side. I'm running `ngrok http 8080` after starting the app with `./gradlew` and I'm using `$NGROK_URL/scim` as the value. Any idea how to fix it?

Screen Shot 2022-10-02 at 10 40 01

**Comment by mraible** [comment]
I'm able to make the connection from Okta work if I remove the API Token value.

Screen Shot 2022-10-02 at 10 51 10

Now if I open http://127.0.0.1:4040/, I can see the request to `/Users`, but it returns a 404. This seems reasonable since my user doesn't exist. I'm not sure why the user isn't created though.

Screen Shot 2022-10-02 at 11 04 51

**Pull request merged**